kit/src/components/message_typing/mod.rs

```diff
@@ -1,24 +1,63 @@
+use common::language::{get_local_text, get_local_text_args_builder};
 use dioxus::prelude::*;
 
-#[derive(Props)]
-pub struct Props<'a> {
+const MAX_LEN: usize = 17;
+
+#[derive(Props, PartialEq)]
+pub struct Props {
     // Represents the image of the user who is typing
-    user_image: Element<'a>,
+    typing_users: Vec<String>,
 }
 
 #[allow(non_snake_case)]
-pub fn MessageTyping<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
+pub fn MessageTyping(cx: Scope<Props>) -> Element {
+    let typing_users = if cx.props.typing_users.len() > 3 {
+        get_local_text("messages.users-multiple-typing")
+    } else if cx.props.typing_users.len() == 1 {
+        let mut user = cx.props.typing_users.first().unwrap().to_string();
+        let user = if user.len() > MAX_LEN {
+            let mut user: String = user.drain(..(MAX_LEN - 3)).collect();
+            user.push_str("...");
+            user
+        } else {
+            user
+        };
+        get_local_text_args_builder("messages.user-typing", |m| {
+            m.insert("user", user.into());
+        })
+    } else {
+        let mut users = cx.props.typing_users.join(", ");
+        let users = if users.len() > MAX_LEN {
+            let mut users: String = users.drain(..(MAX_LEN - 3)).collect();
+            users.push_str("...");
+            users
+        } else {
+            users
+        };
```

Comment:
i think that lines 16-36 could be replaced by lines 29-36.  `join` on an array of length 1 would have no effect.

Reply:
done